File: libraries/cyclestreets-view/src/main/java/net/cyclestreets/liveride/LiveRideStart.java
package net.cyclestreets.liveride;

import net.cyclestreets.routing.Journey;

import org.osmdroid.util.GeoPoint;

import android.speech.tts.TextToSpeech;

final class LiveRideStart extends LiveRideState
{
  LiveRideStart(final LiveRideService liveRideService, final TextToSpeech tts) {
    super(liveRideService, tts);
    notifyAndSetServiceForeground(liveRideService, "Starting LiveRide");
  }

  @Override
  public LiveRideState update(Journey journey, GeoPoint whereIam, int accuracy) {
    journey.setActiveSegmentIndex(0);
    notify(journey.activeSegment());
    return new HuntForSegment(this);
  }

  @Override
  public boolean isStopped() { return false; }
  @Override
  public boolean arePedalling() { return false; }
}

File: libraries/cyclestreets-view/src/main/java/net/cyclestreets/liveride/LiveRideStart.kt
package net.cyclestreets.liveride

import android.app.Service
import android.content.Context
import android.speech.tts.TextToSpeech
import android.util.Log
import net.cyclestreets.routing.Journey
import org.osmdroid.util.GeoPoint

internal class LiveRideStart(context: Context, tts: TextToSpeech?) : LiveRideState(context, tts) {

    fun setServiceForeground(liveRideService: Service): LiveRideStart {
        notifyAndSetServiceForeground(liveRideService, "Starting LiveRide")
        return this
    }

    override fun update(journey: Journey, myLocation: GeoPoint, accuracy: Int): LiveRideState {
        journey.setActiveSegmentIndex(0)
        Log.d("importantTest", "LiveRideStart Update: ${journey.activeSegment()!!.toString()}")
        notify(journey.activeSegment()!!, true)


        return HuntForSegment(this)


    }

    override fun isStopped(): Boolean { return false }
    override fun arePedalling(): Boolean { return false }
}
